lib/Sema/CSDiag.cpp

```diff
@@ -3653,6 +3653,12 @@ addTypeCoerceFixit(InFlightDiagnostic &diag, ConstraintSystem *CS,
   return false;
 }
 
+/// Recursively read through Optional and IUO types to find a type that isn't.
+static Type readThroughOptionalness(Type type) {
```

Comment:
This actually exists already as well: it's called `lookThroughAllAnyOptionalTypes`.